File: src/main/java/io/github/mg138/tsbook/utils/config/gui/element/GUIElementSetting.java
package io.github.mg138.tsbook.utils.config.gui.element;

import org.bukkit.Material;

import java.util.List;

public class GUIElementSetting {
    public final int slot;
    public final Material material;
    public final int count;
    public final String name;
    public final List<String> lore;
    public final int model;

    public GUIElementSetting(int slot, Material material, int count, String name, List<String> lore, int model) {
        this.slot = slot;
        this.material = material;
        this.count = count;
        this.name = name;
        this.lore = lore;
        this.model = model;
    }
}

File: src/main/kotlin/io/github/mg138/tsbook/config/gui/element/GUIElementSetting.kt
package io.github.mg138.tsbook.config.gui.element

import io.github.mg138.tsbook.util.translate.TranslateUtil
import org.bukkit.Material
import org.bukkit.configuration.ConfigurationSection

/*
open class GUIElementSetting(
    val slot: Int,
    val material: Material,
    val count: Int,
    val name: String,
    val lore: List<String>,
    val model: Int
) {
    constructor(slot: Int, section: ConfigurationSection) : this(
        slot,
        Material.valueOf(section.getString("MATERIAL")!!),
        section.getInt("COUNT"),
        TranslateUtil.get(section, "NAME", null),
        TranslateUtil.getList(section, "LORE", null),
        section.getInt("MODEL")
    )
}
 */